tests/ipsec/l2tp3hosts.pm

```diff
@@ -0,0 +1,194 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Basic setup and functional L2TPv3 connectivity test.
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+package l2tp3hosts;
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use lockapi;
+use network_utils;
+use Kernel::net_tests qw(
+  add_ipv4_addr
+  add_ipv4_route
+);
+
+sub run_left {
+    my ($self, $setup) = @_;
+    my $dev = iface(0);
+
+    add_ipv4_addr(ip => $setup->{left_ip4}, dev => $dev);
+    add_ipv4_route(
+        dst => "$setup->{right_ip4}/32",
+        via => $setup->{middle_ip4_01}
+    );
+
+    script_retry("ping -c 1 $setup->{middle_ip4_01}", retry => 5);
+
+    # Create the L2TPv3 tunnel
+    assert_script_run(
+        "ip l2tp add tunnel tunnel_id 2001 peer_tunnel_id 2002 " .
+          "encap ip local $setup->{left_ip4} remote $setup->{right_ip4}"
+    );
+
+    # Create the L2TPv3 session
+    assert_script_run(
+        "ip l2tp add session name l2tp4 tunnel_id 2001 " .
+          "session_id 3001 peer_session_id 3002"
+    );
```

Comment:
I suppose we can create function for create L2TPv3 tunnel and session and can be used in run_left and run_right?